contracts/finance/andromeda-validator-staking/src/contract.rs

```diff
@@ -174,7 +185,12 @@ fn execute_claim(
 
     // Check if the validator is valid before unstaking
     is_validator(&deps, &validator)?;
-    let recipient = recipient.unwrap_or(info.sender);
+
+    let recipient = if let Some(recipient) = recipient {
```

Comment:
> Can be written as
> 
> ```rust
> recipient.map(|r| r.get_raw_address(&deps.as_ref())?).unwrap_or(info.sender)
> ```
> 
> I think?

`r.get_raw_address(&deps.as_ref())?` does not throw error in case `get_raw_address` returns `ContractError`. As a result, `map` will return `Result<Addr, ContractError>` and `unwrap_or(info.sender)` will be invalid. We can use `Some(info.sender)` instead and `unwrap` again but in that case I am afraid that origin error thrown by the `get_raw_address` will be gone.

Reply:
Was my concern and reason I added "I think". Leave it as is then!